release.py

```diff
@@ -0,0 +1,143 @@
+class bcolors:
+    HEADER = "\033[95m"
+    OKBLUE = "\033[94m"
+    OKCYAN = "\033[96m"
+    OKGREEN = "\033[92m"
+    WARNING = "\033[93m"
+    FAIL = "\033[91m"
+    ENDC = "\033[0m"
+    BOLD = "\033[1m"
+    UNDERLINE = "\033[4m"
+
+
+parser = argparse.ArgumentParser()
+parser.add_argument(
+    "--clean", action="store_true", help="Cancel existing release process, clean files"
+)
+parser.add_argument(
+    "--ignore-deps", action="store_true", help="Ignore dependency checks"
+)
+parser.add_argument("--fin", action="store_true", help="Finish release proccess")
+parser.add_argument("--sign", type=str, help="Signing key")
+parser.add_argument("--ticket", type=int, help="Ticket number", required=True)
+parser.add_argument("--tag", type=str, help="Release tag", required=True)
+args = parser.parse_args()
+
+
+TAG = args.tag
+TICKET = args.ticket
+TODAY = datetime.date.today()
+BRANCH = "{ticket}.release-{tag}".format(
+    ticket=TICKET, tag=TAG
+)  # looks like XXXX.release-1.16.0
+RELEASE_TITLE = "Release {tag} ({date})".format(tag=TAG, date=TODAY)
+RELEASE_FOLDER = "../tahoe-release-{0}".format(TAG)
+CONTINUE_INSTRUCTION = (
+    bcolors.BOLD
+    + "Instruction : run ./venv/bin/python release.py --ignore-deps --tag {tag} --ticket {ticket} --sign {sign} --fin".format(
+        tag=TAG, ticket=TICKET, sign="YOUR_SIGNING_KEY_HERE"
+    )
+    + bcolors.ENDC
+)
+
+
+def clean():
+    for release_dir in glob.glob("../tahoe-release-*"):
+        shutil.rmtree(release_dir, ignore_errors=True)
+
+
+def check_dependencies():
+    repo_clean = subprocess.run(["git", "status", "-s"], capture_output=True)
+    if repo_clean.stdout:
+        print(
+            f"{bcolors.WARNING}Warning: repo is not clean, please commit changes!{bcolors.ENDC}"
+        )
+        sys.exit(1)
+    try:
+        import wheel
+    except ModuleNotFoundError:
+        print(
+            f"{bcolors.WARNING}Warning: wheel is not installed. Install via pip!{bcolors.ENDC}"
+        )
+        sys.exit(1)
+
+
+def start_release():
+    subprocess.run(
+        ["git", "clone", "https://github.com/tahoe-lafs/tahoe-lafs.git", RELEASE_FOLDER]
+    )
+    os.chdir(RELEASE_FOLDER)
+    subprocess.run(["python", "-m", "venv", "venv"])
+    subprocess.run(["./venv/bin/pip", "install", "--editable", ".[test]"])
+
+    subprocess.run(["git", "branch", BRANCH])
+    subprocess.run(["git", "checkout", BRANCH])
+    subprocess.run(["./venv/bin/tox", "-e", "news"])
+    subprocess.run(["touch", "newsfragments/{ticket}.minor".format(ticket=TICKET)])
+    subprocess.run(["git", "add", "."])
+    subprocess.run(
+        ["git", "commit", "-s", "-m", "tahoe-lafs-{tag} news".format(tag=TAG)]
+    )
+    with open("NEWS.rst", "r+") as f:
+        content = f.read()
+        updated_content = re.sub(
+            r"\.\.\stowncrier start line\n(Release\s(\d+\.\d+\.\d)(\.)post\d+\s\(\d{4}-\d{02}-\d{02}\))+(\n\'+)",
+            "",
+            content,
+        )
+        f.write(updated_content)
```

Comment:
You either want to seek to the start of the file before rewriting it and then truncate it afterwards or write your output into temporary file and rename it over the original.
The latter approach is more resilient to crashes and restarts and also allows for line by line processing which would help if we were dealing with huge file, but that is not really the case.

Reply:
Thanks, that worked.